### DISCUSSION TITLE
improve typescript compilation times with snapshots

### DISCUSSION THREAD

**Discussion by andykais** [denoland/deno#14736] [Ideas]
so the typescript repo has an issue discussing speeding up the compiler, I thought I would share that issue here https://github.com/microsoft/TypeScript/issues/25658#issuecomment-1138877141. Especially of note, is that node 18 supports user-land snapshotting https://nodejs.org/en/blog/announcements/v18-release-announce/#build-time-user-land-snapshot-experimental, and the effort appears to be mostly focused on speeding up the compiler using snapshots.

This is an interesting feature, and I can imagine use cases outside the tsc compiler, but this seems like the most immediately valuable place to use snapshotting.

As far as I know, deno already does some snapshotting of user code to improve startup times. It could be valuable to do the same for the typescript compiler. Please correct me if this is something deno already does 😅

**Comment by kitsonk** [comment]
We already snapshot the compiler.  It was one of the early optimisations we did a long time ago.

You can see the code here in the build process: https://github.com/denoland/deno/blob/main/cli/build.rs#L395

**Reply by andykais** [reply]
oh awesome. I guess this discussion is closed then lol. Deno was way ahead of node!
